src/geometry/lane_geometry.cc

```diff
@@ -96,9 +96,17 @@ maliput::math::Vector3 LaneGeometry::WDot(const maliput::math::Vector3& prh) con
 }
 
 maliput::math::RollPitchYaw LaneGeometry::Orientation(double p) const {
-  const double superelevation{0.}; /* TODO: Take superelevation into account */
+  const maliput::math::Vector3 on_left = ToLateralPos(LineStringType::kLeftBoundary, p);
+  const maliput::math::Vector3 on_right = ToLateralPos(LineStringType::kRightBoundary, p);
+  const maliput::math::Vector2 on_left_2d{on_left.x(), on_left.y()};
+  const maliput::math::Vector2 on_right_2d{on_right.x(), on_right.y()};
+  const double xy_distance = (on_left_2d - on_right_2d).norm();
+  const double elevation_diff = on_left.z() - on_right.z();
+  MALIPUT_THROW_UNLESS(xy_distance != 0.);
+
+  const double superelevation{elevation_diff / xy_distance};
```

Comment:
This is the arc tangent argument, not the superelevation itself. Prefer instead to use std::atan2 and let it solve any of the 0°, 90°, 180° and 270° cases of the argument for you.

Reply:
You are right. Superelevation by definition is exactly the roll angle.